### PR TITLE
Thread the logger through the apicaller.

### PR DESCRIPTION
**Pull request by howbazaar**
Pass a logger into the apicaller manifold config, and use that logger internally.

### CODE REVIEW

File: cmd/jujud/agent/caasoperator/manifolds.go [hunk 1]
```diff
@@ -130,6 +130,7 @@ func Manifolds(config ManifoldsConfig) dependency.Manifolds {
 			APIOpen:              api.Open,
 			APIConfigWatcherName: apiConfigWatcherName,
 			NewConnection:        apicaller.OnlyConnect,
+			Logger:               loggo.GetLogger("juju.worker.apicaller"),
```
Comment (babbageclunk) [inline cmd/jujud/agent/caasoperator/manifolds.go]: So this one will still be a package level one? Oh, I guess this worker isn't model specific?

File: cmd/jujud/agent/model/manifolds.go [hunk 2]
```diff
@@ -145,6 +145,7 @@ func commonManifolds(config ManifoldsConfig) dependency.Manifolds {
 			APIOpen:       api.Open,
 			NewConnection: apicaller.OnlyConnect,
 			Filter:        apiConnectFilter,
+			Logger:        config.LoggingContext.GetLogger("juju.worker.apicaller"),
```
Comment (babbageclunk) [inline cmd/jujud/agent/model/manifolds.go]: Right, and I guess this is the key part that gets the model-specific logger for this case.

Comment (howbazaar) [reply cmd/jujud/agent/model/manifolds.go]: Yup

**Comment by howbazaar** [comment]
$$merge$$

**Pull request merged**